src/vellum/workflows/graph/graph.py

```diff
@@ -42,6 +42,13 @@ def from_port(port: "Port") -> "Graph":
     @staticmethod
     def from_node(node: Type["BaseNode"]) -> "Graph":
         ports = {port for port in node.Ports}
+        if not ports:
+            from vellum.workflows.ports.port import Port
+
+            default_port = Port(default=True)
```

Comment:
Instead of a default port, create a new class in this file called `NoPortsNode` which simply wraps the node class. The `terminals` field can accept this class, and update edge creation logic such that an edge can't be from from a NoPortsNode to the next node